src/capi.rs

```diff
@@ -444,6 +444,26 @@ pub unsafe extern fn mp4parse_get_track_video_info(parser: *mut mp4parse_parser,
     MP4PARSE_OK
 }
 
+#[no_mangle]
+pub unsafe extern fn mp4parse_is_fragmented(parser: *mut mp4parse_parser, track_id: u32, fragmented: *mut bool) -> mp4parse_error {
+    if parser.is_null() || (*parser).poisoned() {
+        return MP4PARSE_ERROR_BADARG;
+    }
+
+    let context = (*parser).context_mut();
+    let tracks = &context.tracks;
+    (*fragmented) = false;
+
+    for track in tracks {
+        if track.track_id.unwrap() == track_id && track.fragmented == true {
```

Comment:
Please don't call `unwrap()` in library code. It will crash the application, so this is at least as strong as `MOZ_RELEASE_ASSERT`. Instead check `track.track_id.is_none()` or use a `match` and skip it or return MP4PARSE_ERROR_INVALID depending the behaviour you  want.

Calling `unwrap()` in test code is fine since it's a good way to abort on unexpected state.